docs/extensions/custom-bootstrapping.md

```diff
@@ -1,89 +1,169 @@
+or `.bolt.php`. 
+
+By default, Bolt uses `.bolt.yml` because who doesn't love YAML? But, for more
+complex, or dynamic configuration options, you can use a PHP file instead, 
+`.bolt.php`.
+
+The basics of configuring a Bolt application
+--------------------------------------------
+
+The minimal requirement of configuring Bolt's bootstrap is path configuration. 
+
+Bolt needs to know these paths before configuration files like `config.yml` are
+read, in order to know where to find them.
+
+A default Bolt install will have a `.bolt.yml` file that would look similar to:
+
+```yaml
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
 ```
 
-Ideally you would replace the path to bootstrap.php with a link to your own
-bootstrap file, by convention it is best to name this `.bolt.php` since that filename
-is looked for automatically by other Bolt components. For example:
+The equivalent PHP file to the one above would look like:
 
+```php
+<?php
+
+return [
+    'paths' => [
+        'web'       => 'public',
+        'themebase' => 'public/theme',
+        'files'     => 'public/files',
+        'view'      => 'public/bolt-public/view',
+    ]
+];
 ```
-$app = require __DIR__ . '/../.bolt.php';
-if ($app === false) {
-    return false;
-}
-$app->run();
+
+Custom Application Class
+------------------------
+
+The Application class can be changed here as well. This allows you to modify
+Bolt services or add your own.
+
+In YAML, you can give the class name and we will create it for you. We will
+still apply any path customization afterwards as well. 
+
+```yaml
+application: My\Application
+paths:
+    web: public
+    #...
 ```
 
-The basics of configuring a Bolt application
---------------------------------------------
+In PHP, you can give the class name...
 
-The job of your custom bootstrap file is to provide a bootstrapped `$app`
-object that the `index.php` file will then run. The simplest possible bootstrap
-file will look like this, assuming that your bootstrap file is in the 
-project root one level down from your `index.php` file in `./public`:
+```php
+<?php
 
+return [
+    'application' => My\Application::class,
+    'paths' => [
+        'web' => 'public'
+        //...
+    ]
+];
 ```
-// .bolt.php
-$configuration = new Bolt\Configuration\Composer(__DIR__);
-$configuration->setPath("web", "public");
 
-$app = new Bolt\Application(array('resources'=>$configuration));
+For even more flexibly you can create the application yourself, or pull it in
+from another bootstrap file you have. It’s important to note here that if you
+give us an application object we'll assume it is ready to go. This means you'll
+need to apply any path customizations yourself. See the next section for an 
+example.
 
-$app->initialize();
-$config = [
-    'application' => $app,
-    'resources' => null,
+```php
+<?php
+
+// However you need create the application.
```

Comment:
Nah, just me not understanding the shit and trying to give the sentence at least a bit of sense :P